lib/oelite/cmd/check_provided.py

```diff
@@ -0,0 +1,182 @@
+import oebakery
+import oelite.baker
+import logging
+from oelite.parse import confparse
+import re
+import subprocess
+import bb.utils
+
+description = "Do version check of provided utils"
+
+def add_parser_options(parser):
+    oelite.baker.add_show_parser_options(parser)
+    parser.add_option("-v", "--verbose",
+                      action="store_true",
+                      help="Be more chatty")
+    parser.add_option("-s", "--show",
+                      action="store_true",
+                      help="Detect and print provided versions, do not compare to expected")
+    return
+
+def parse_args(options, args):
+    if options.verbose:
+        logging.getLogger().setLevel(logging.DEBUG)
+    else:
+        logging.getLogger().setLevel(logging.INFO)
+
+INITIAL_OE_IMPORTS = "sys os time"
+
+# One must describe how to get the version of each util, along with
+# the expected minimal and maximal versions. Fortunately, many
+# utilities support a simple "--version" flag, and for most, the
+# command to call is the same as what's in the ASSUME_PROVIDED (an
+# exception is native:pkgconfig, where we the command is
+# pkg-config). So we don't need to be very verbose.
+#
+# extract: a regexp string which is matched against the output from
+# the command. Should contain a single capture group, from which we'll
+# get the version string
+#
+# min, max: obvious. Both are optional (though if both are omitted one
+# might as well not have this...).
+#
+# ret: Some utilities exit with a non-zero value when asked to print
+# their version (sometimes because they don't actually explicitly
+# support --version or any other flag, but they display their version
+# nevertheless as a side effect of telling the user he did something
+# wrong...). This can be set to an integer or list of integers of
+# expected return values.
+data = dict()
+data["binutils"] = dict(cmd = "ld")
+data["coreutils"] = dict(cmd = "ls")
+data["mercurial"] = dict(cmd = "hg")
+data["mtd-utils-mkfs-jffs2"] = dict(cmd = "mkfs.jffs2", ret = 255)
+data["perl"] = dict(extract = r"\(v([0-9][0-9.a-zA-Z_-]+)\)")
+data["pkgconfig"] = dict(cmd = "pkg-config")
+data["python-runtime"] = dict(cmd = "python")
+data["texinfo"] = dict(cmd = "texindex")
+data["unzip"] = dict(args = "-v")
+data["util-linux"] = dict(cmd = "getopt")
+
+data["git"] = dict(min = "1.8.4")
```

Comment:
it doesn't seem like a good idea to add version information here, since I don't think everybody would know they need to update a sub-command if the touch provided.conf :)

Would it be a problem to add this information to the .conf file instead?